magma/generator.py

```diff
@@ -71,24 +72,26 @@ def _make_key(cls, *args, **kwargs):
 def _make_type(cls, *args, **kwargs):
     dummy = type.__new__(cls, "", (), {})
     name = cls.__name__
-    bases = (Circuit,)
-    dct = DefineCircuitKind.__prepare__(name, bases)
+    bases = (cls._base_cls,)
+    dct = cls._base_kind.__prepare__(name, bases)
```

Comment:
kind -> metacls

Reply:
and let's use `sunderscore` for DSL stuff per caleb, so `_base_cls_`, `_base_metacls_`